src/pages/ContentScripts/features/developer-hovercard-info/view.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from 'react';
+import ReactDOM from 'react-dom';
+import getGithubTheme from '../../../../helpers/get-github-theme';
+import optionsStorage, { HypercrxOptions, defaults } from '../../../../options-storage';
+import { useTranslation } from 'react-i18next';
+import '../../../../helpers/i18n';
+import { rocketLight, rocketDark } from './base64';
+
+interface OpenRankProps {
+  developerName: string;
+  openrank: string;
+}
+
+const OpenRankView: React.FC<OpenRankProps> = ({ developerName, openrank }) => {
+  const [theme, setTheme] = useState<'light' | 'dark'>(getGithubTheme() as 'light' | 'dark');
+  const [options, setOptions] = useState<HypercrxOptions>(defaults);
+  const { t, i18n } = useTranslation();
+  useEffect(() => {
+    (async function () {
+      setOptions(await optionsStorage.getAll());
+      i18n.changeLanguage(options.locale);
+    })();
+  }, [options.locale]);
+
+  useEffect(() => {
+    const handleThemeChange = () => {
+      const newTheme = getGithubTheme() as 'light' | 'dark';
+      setTheme(newTheme);
+    };
+
+    // Listen for theme change events
+    window.addEventListener('themeChange', handleThemeChange);
+
+    return () => {
+      window.removeEventListener('themeChange', handleThemeChange);
+    };
+  }, []);
```

Comment:
Why do we add a listener here and remove it? For most of our other features, we have only a line of `const githubTheme = getGithubTheme();` to handle theme. Let me know if this is not working.

Reply:
already edited